### PR TITLE
Docker to ease local development

### PR DESCRIPTION
**Pull request by dulacp**
Three docker containers were needed to make local development easier and deterministic no matter which OS you are using: backends, frontend, mysql.

The python containers are based on [Alpine Linux](https://en.wikipedia.org/wiki/Alpine_Linux), to minimize the finale size (Alpine Linux has a size of only 4MB!).

Then, `docker-compose` has been used to orchestrate the build/creation and starting of the containers.

I've added instructions in the bottom of the README, basically once you have a GCP project with a service-account configured you only have to run:

1. once to build the containers: 
    ```sh
    $ docker-compose build
    ```
1. start the containers:
    ```sh
    $ export APPLICATION_ID=$PROJECT_ID
    $ docker-compose up
    ```

NB: You can [install docker for mac from the Docker Store](https://store.docker.com/editions/community/docker-ce-desktop-mac).

Cheers

### CODE REVIEW

File: docker-compose.yaml [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    depends_on:
+      - db
+      - migrations
+
+  frontend:
+    build:
+      context: .
+      dockerfile: docker/frontend/Dockerfile
+    ports:
+      - "4200:4200"
```
Comment (Slony) [inline docker-compose.yaml]: Please, add the following two lines here to make frontend as responsive to file changes as backend is:

```yaml
    volumes:
      - ./frontend/src:/app/frontend/src
```

Comment (dulacp) [reply docker-compose.yaml]: Good call! :)

File: docker-compose.yaml [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+      dockerfile: docker/backends/Dockerfile
+    environment:
+      DATABASE_URI: mysql+mysqldb://crmintapp:crmintapp@db:3306/crmintapp
+    command: >
+      bash -c "export PYTHONPATH="/root/google-cloud-sdk/platform/google_appengine:lib" &&
+               export FLASK_APP=run_ibackend.py &&
+               export FLASK_DEBUG=1 &&
+               export APPLICATION_ID=crmintapp &&
+               cd /app/backends &&
+               python -m flask db_seeds"
```
[the remark this answers is not in the document]
Comment (dulacp) [reply docker-compose.yaml]: Thanks for catching that!

File: backends/gae_dev_ibackend.yaml.example [hunk 3]
```diff
@@ -1,25 +0,0 @@
-service: api-service
```
[the remark this answers is not in the document]
Comment (dulacp) [reply backends/gae_dev_ibackend.yaml.example]: Why not just add the `gae_dev_ibackend.yaml` and `gae_dev_jbackend.yaml` files?

**Pull request merged**